app/views/application/_login_items.html.haml

```diff
@@ -1,5 +1,17 @@
 - if user_signed_in?
-  = link_to(t('devise.sessions.destroy.log_out'), destroy_user_session_path, method: :delete)
+  %a.dropdown-toggle.link-no-styling.tall-line{ href: '#',
+     'data-toggle' => 'dropdown', role: 'button' }
+    = t('hello', name: current_user.first_name)
+    %span.caret
+
+  %ul.dropdown-menu
+    %li
+      = link_to(t('devise.registrations.edit.title'),
+                edit_user_registration_path(current_user))
```

Comment:
I think it would be better to remove the `current_user` argument. As it stands, it creates a `/users/edit.1` instead of a `/users/edit` link. And I'm not sure why, but on my system that `/users/edit.1` url takes **much** longer to render than the regular link.